### PR TITLE
Fix emptydrops Bug

### PR DESCRIPTION
**Pull request by barkasn**
### Purpose

- Fix empty drops bug

### Changes

- Fix empty drops bug
- Update ubuntu image for that step
- Add parameter to Optimus that allows setting parameters now required for emptyDrops to pass tests
- Update tests with checksums
- Add a new test directory for 4k PBMC (this does not run as part of PR tests, but is standard test we use for a full dataset run), I had to use it to verify pipeline and its very helpful so included it
- Fix exporting of emptyDrops to ZARR

### Review Instructions

- No instructions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -182,7 +182,7 @@ workflows:
           - test_optimus_mouse
           - test_smartseq2
           - test_smartseq2_single_end
-          - test_emptyDropsWrapper
+#          - test_emptyDropsWrapper
```
Comment (pshapiro4broad) [inline .circleci/config.yml]: I think it would be better to delete this and create a JIRA in our backlog to track adding this test back.

File: docker/zarr-output/Dockerfile [hunk 2]
```diff
@@ -2,8 +2,11 @@ FROM python:3.7.2
 
 LABEL maintainer="Lantern Team <lantern@broadinstitute.org>"
 
+RUN pip install --upgrade pip
+
 COPY requirements.txt .
 RUN pip3 install numpy==1.17.0
+RUN pip3 install cython
```
Comment (pshapiro4broad) [inline docker/zarr-output/Dockerfile]: add version for cython

File: pipelines/optimus/Optimus.wdl [hunk 3]
```diff
@@ -46,6 +46,9 @@ workflow Optimus {
   # If true produce the optional loom output
   Boolean output_loom = false
 
+  # Emptydrops lower cutoff
+  Int emptydrops_lower = 100
```
Comment (pshapiro4broad) [inline pipelines/optimus/Optimus.wdl]: Since this is a new parameter, make sure the documentation is updated

File: test/optimus/4kpbmc/ValidateOptimus.wdl [hunk 4]
```diff
@@ -0,0 +1,272 @@
+workflow ValidateOptimus {
```
Comment (pshapiro4broad) [inline test/optimus/4kpbmc/ValidateOptimus.wdl]: When/if we migrate to a new repo, we need to figure out a way to do this without code duplication.

File: test/optimus/4kpbmc/test_inputs.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "TestOptimusPR.expected_bam_hash": "83234b3b714c96d897b44d91169933bd",
+  "TestOptimusPR.reference_matrix": "gs://hca-dcp-sc-pipelines-test-data/optimus-test-data/referenceMatrix.rds",
+  "TestOptimusPR.expected_cell_metric_hash": "eb56bfec00cb35ea176c00cebc44c4e3",
```
Comment (pshapiro4broad) [inline test/optimus/4kpbmc/test_inputs.json]: need to make sure these checksums are correct, and/or remove them and make sure that the validation doesn't require them.

File: test/optimus/pr/test_inputs.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
   "TestOptimusPR.reference_matrix": "gs://hca-dcp-sc-pipelines-test-data/optimus-test-data/referenceMatrix.rds",
   "TestOptimusPR.expected_cell_metric_hash": "8bc4a2e96b6404e47f071aa1c06d196f",
   "TestOptimusPR.expected_gene_metric_hash": "c8f0cd1714d08d43bfbd38c6b353b7f9",
-  "TestOptimusPR.expected_loom_file_checksum": "590cf3bf96fb5323adbfad60d635abb2",
+    "TestOptimusPR.expected_loom_file_checksum": "3e4ee79c570cb3e4fc59a26840eaf785",
```
Comment (pshapiro4broad) [inline test/optimus/pr/test_inputs.json]: unnecessary whitespace change
```suggestion
   "TestOptimusPR.expected_loom_file_checksum": "3e4ee79c570cb3e4fc59a26840eaf785",
```

File: test/optimus/pr/test_inputs.json [hunk 7]
```diff
@@ -22,5 +22,5 @@
   "TestOptimusPR.tar_star_reference": "gs://hca-dcp-sc-pipelines-test-data/alignmentReferences/optimusGencodeV27/buildReference/output_bucket/star_primary_gencode_v27.tar",
   "TestOptimusPR.sample_id": "pr_test_dataset",
   "TestOptimusPR.annotations_gtf": "gs://hca-dcp-sc-pipelines-test-data/alignmentReferences/optimusGencodeV27/gencode.v27.primary_assembly.annotation.gtf.gz",
-  "TestOptimusPR.ref_genome_fasta": "gs://hca-dcp-sc-pipelines-test-data/alignmentReferences/optimusGencodeV27/GRCh38.primary_assembly.genome.fa"
+  "TestOptimusPR.ref_genome_fasta": "gs://hca-dcp-sc-pipelines-test-data/alignmentReferences/optimusGencodeV27/GRCh38.primary_assembly.genome.fa",
```
Comment (pshapiro4broad) [inline test/optimus/pr/test_inputs.json]: remove trailing comma
```suggestion
  "TestOptimusPR.ref_genome_fasta": "gs://hca-dcp-sc-pipelines-test-data/alignmentReferences/optimusGencodeV27/GRCh38.primary_assembly.genome.fa"
```

File: .circleci/config.yml (hunk 1 above)
Comment (barkasn) [reply .circleci/config.yml]: GL-854 created

File: pipelines/optimus/Optimus.wdl (hunk 3 above)
Comment (barkasn) [reply pipelines/optimus/Optimus.wdl]: I've checked and we don't document to this detail. As this is a parameter with a default value that will never be changed by the end user I also don't think it needs to be added to the default inputs file.

File: test/optimus/4kpbmc/test_inputs.json (hunk 5 above)
Comment (barkasn) [reply test/optimus/4kpbmc/test_inputs.json]: fixed
